### ISSUE TITLE
[core] Gradle build generates Incremental Analysis warning

### ISSUE THREAD

**Issue by AdamCarroll** [pmd/pmd#921]
**Affects PMD Version:** 

6.0.1

**Rule:**

N/A

**Description:**

Running a Gradle build with PMD 6.0.1 generates the following warning:

> This analysis could be faster, please consider using Incremental Analysis: https://pmd.github.io/pmd-6.0.1/pmd_userdocs_getting_started.html#incremental-analysis

There does not seem to be a way to fix the problem and thus silence the warning.

**Code Sample demonstrating the issue:**

For example, launching the build on the [VocabHunter project](https://github.com/VocabHunter/VocabHunter) shows the problem:

```
$ ./gradlew clean :core:pmdMain

> Task :core:pmdMain 
This analysis could be faster, please consider using Incremental Analysis: https://pmd.github.io/pmd-6.0.1/pmd_userdocs_getting_started.html#incremental-analysis


BUILD SUCCESSFUL in 10s
8 actionable tasks: 3 executed, 5 up-to-date
```

**Running PMD through:** *Gradle*

**Comment by jsotuyod** [comment]
@AdamCarroll hello and welcomed!

Unfortunately this is not an issue with PMD itself, but with how Gradle exposes PMD to users through it's DSL.

I submitted a PR to support configuring analysis cache back in ~December. They merged it, then rollbacked it until we released 6.0.0 (they wanted extra logs messages so they can test PMD is using it's own cache correctly), and have ever since been saying they would bring it back.

You can let them know you are interested in this happening through their own github project: https://github.com/gradle/gradle/pull/3125

In the meantime, other than being noisy, the log is harmless (pretty much like Gradle's own message about not using the daemon back in Gradle 2.x)

**Comment by AdamCarroll** [comment]
Hi @jsotuyod .  Thanks for the information.  Well done for submitting that pull request.  Hopefully it will make its way into a public release of Gradle soon.

Keep up the good work on PMD.  It's a useful tool!

**Comment by kreiger** [comment]
I downgraded to PMD 5.8.1, which got rid of the warning.

**Comment by AdamCarroll** [comment]
That's certainly one way round it @kreiger .  I'm still hoping that the PR from @jsotuyod will get re-merged to solve the problem and avoid that.

**Comment by jsotuyod** [comment]
That's missing almost a whole year of development, and 5 non-bugfixing releases...

**Comment by AdamCarroll** [comment]
Not much hope then.  It's a shame to have to take the route of downgrading.

**Comment by kreiger** [comment]
I'm working on making the build output as silent as possible, and unfortunately i couldn't find any other way to silence the warning.

**Comment by AdamCarroll** [comment]
Yep, that's a good way to avoid accidentally missing real problems that can sometimes happen when the build output is too noisy.

**Comment by eatdrinksleepcode** [comment]
@jsotuyod on 7/3/2018 @blindpirate [said](https://github.com/gradle/gradle/pull/3125#issuecomment-402165303) that they won't be able to get to it, but could review a submission. Any chance that you have bandwidth to do that? If not, do you think someone with plenty of Java and Gradle experience but not much with PMD could take it on?

**Comment by jsotuyod** [comment]
@eatdrinksleepcode I do not have the bandwidth at the moment, as I already explained; but definitely someone else can pick it up.

The code itself existed, and you would basically have to revert https://github.com/gradle/gradle/pull/3125, and make sure everything still works (several Gradle versions shipped in between, and for Gradle 5 they are bumping the default PMD version to 6.6.0 or 6.7.0, I don't recall).

Additionally, support for `-nocache` could be added, but that's not as important.

**Comment by AdamCarroll** [comment]
Do you think it's worth reopening this issue since the problem still exists? It's a shame that the Gradle fix didn't get included but from what I read above it's too late for that now so it probably needs some new work from someone.

**Comment by Cinderhaze** [comment]
This looks like it was resolved in gradle 5.6 - https://docs.gradle.org/5.6/release-notes.html#support-for-pmd-incremental-analysis - It was merged in commit https://github.com/gradle/gradle/commit/9093bb01eeecda38e9ec8465295c9497b6ff1924
